runtime-jvm/main/src/main/scala/Builtins.scala

```diff
@@ -11,33 +11,78 @@ import org.unisonweb.util.{Sequence, Stream, Text}
 
 /* Sketch of convenience functions for constructing builtin functions. */
 object Builtins {
+  def env =
+    (new Array[U](20), new Array[B](20), StackPtr.empty, Result())
 
   // Stream.empty : Stream a
   val Stream_empty =
-    c0l("Stream.empty", Stream.empty[Value])
+    c0z("Stream.empty", Stream.empty[Value])
 
 
-  // Stream.fromInt : Int64 -> Stream Int64
-  val Stream_fromInt = // Stream.iterate(unison 0)(Int64_inc)
-    fp_l("Stream.fromInt", "n", Stream.fromUnison)
+  // Stream.fromInt : Integer -> Stream Integer
+  val Stream_fromInt = // Stream.iterate(unison 0)(Integer_inc)
+    fp_z("Stream.fromInt", "n", Stream.fromUnison)
 
   // Stream.cons : a -> Stream a -> Stream a
   val Stream_cons =
-    fpp_l("Stream.cons", "v", "stream",
+    fpp_z("Stream.cons", "v", "stream",
           (v: Value, stream: Stream[Value]) => v :: stream)
 
+  val Stream_take =
+    flp_z("Stream.take", "n", "stream",
+          (n, s: Stream[Value]) => s.take(n))
+
   val Stream_drop =
     flp_z("Stream.drop", "n", "stream",
           (n, s: Stream[Value]) => s.drop(n))
 
-  // Stream.map : Stream a -> (a -> b) -> Stream b
-  //  val Stream_map = fpp_p("Stream.map", "stream", "f", (s: Stream[Value], b: Value) => ???)
+  // Stream.map : (a -> b) -> Stream a -> Stream b
+  val Stream_map =
+    fpp_z("Stream.map", "f", "stream",
+          (f: Value, s: Stream[Value]) =>
+            s.map(UnisonToScala.toUnboxed1(f.asInstanceOf[Lambda])(env)))
+
+  // Stream.foldLeft : b -> (b -> a -> b) -> Stream a -> b
+  val Stream_foldLeft =
+    fppp_p("Stream.foldLeft", "acc", "f", "stream",
+           (acc: Value, f: Value, s: Stream[Value]) =>
+             s.foldLeft(acc)(
+               UnisonToScala.toUnboxed2(f.asInstanceOf[Lambda])(env))
+           // todo: env needs to be bigger here
```

Comment:
I don't remember exactly what I meant by "bigger" here; bigger than 20, maybe?  But I do find it a little sketchy that I'm allocating new stacks here and there.